Update Smoke Testing advice

DanielMurphy22 opened the pull request:
**Description of work.**


**To test:**

<!-- Instructions for testing. -->

Fixes #33519 the points raised in the second comment: https://github.com/mantidproject/mantid/issues/33519#issuecomment-1033710455

Read the documentation updates and check that they clearly address the points raised in the issue.

<!-- Ensure the base of this PR is correct (e.g. release-next or master)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

DanielMurphy22 committed:
Update Smoke Testing advice

sf1919 commented:
One thing I find extremely confusing about this page is the use of headings such as 'Monday, 3 weeks' and 'Thursday, 2 days'. What do they mean?

DanielMurphy22 commented:
> One thing I find extremely confusing about this page is the use of headings such as 'Monday, 3 weeks' and 'Thursday, 2 days'. What do they mean?

This never made sense to me but now I've added an update, I think it means time away from "the release day". So we Manual text with x weeks left and smoke test with 1 day left. I agree it's not clear. Not sure what would be better?

sf1919 commented:
Could we not just say '3 weeks before release due date' or something similar and remove the days of the week? I think a timeline is useful but it's the days that are causing the confusion.

sf1919 commented on dev-docs/source/ReleaseChecklist.rst:
```diff
@@ -110,21 +110,46 @@ with the Release Manager.
+Smoke Testing
+^^^^^^^^^^^^^
+
+*  Make sure to follow the preparing steps listed above for yesterday.
```
```suggestion
*  Make sure to follow the preparation steps listed above for yesterday.
```

DanielMurphy22 commented:
> Could we not just say '3 weeks before release due date' or something similar and remove the days of the week? I think a timeline is useful but it's the days that are causing the confusion.

@thomashampson @martyngigg (@sf1919) I thought I'd involve your opinions before making any changes. As the 4 of us are the ones involved in this page I think we should make it what we want! I'll admit I've used this page less than I probably should have in the past, but I agree with Sarah that at least initially these titles such as:
`Wednesday, 2 weeks & 3 days` are confusing. 

One options is to simply add `left` or `until release` to these titles. We could also remove the day of the week, unless you feel it adds value for your usual role? Open to suggestions!

DanielMurphy22 committed:
Edit release timeline

martyngigg commented:
I must admit to just being used to it and having done this too much that I don't actually look at the pages. The reorg by role was an improvement. Perhaps what would help would be defining a list of "events", such as feature freeze, start of beta testing etc and these are listed at the top. I'd say a diagram but they can be a pain to keep up to date. This way as the events may move in time the doc doesn't need to change - we just check locally what dates are set.

sf1919 commented:
I think having a list of events in order of when they need to occur is probably a better way forward and as @martyngigg says it's more flexible and less likely to need changing in the future.

DanielMurphy22 committed:
Update Release Checklist with QA Smoke testing preparation

Reorient timeline by Key Events
Reword all roles to people (apart from Release Manager)
Follow Mantid DocumentationGuideForDevs

sf1919 commented on dev-docs/source/ReleaseChecklist.rst:
```diff
@@ -12,131 +12,146 @@ request or perform a patch release look at the
+* Send an invite to developers for 1.5 hours maximum Smoke Testing. Include an introduction message to assign all testers to a certain operating system.
+  Link to the release pipeline builds where the release packages *WILL* be. Encourage testers to download
+  in the 30 mins before smoke testing. Inform that ticking on a testing issue means that someone has assigned themselves and will tackle that task.
+* The QA manager should pre-setup 3 ISIS IDAaaS mantid dev instances and manually install the release package before testing
```
```suggestion
* The QA manager should pre-setup 3 ISIS IDAaaS Mantid dev instances and manually install the release package before testing
```

sf1919 commented on dev-docs/source/ReleaseChecklist.rst:
```diff
@@ -12,131 +12,146 @@ request or perform a patch release look at the
+* Liaise with the technical release manager and project manager to decide on an appropriate time for Smoke Testing.
+* Send an invite to developers for 1.5 hours maximum Smoke Testing. Include an introduction message to assign all testers to a certain operating system.
+  Link to the release pipeline builds where the release packages *WILL* be. Encourage testers to download
+  in the 30 mins before smoke testing. Inform that ticking on a testing issue means that someone has assigned themselves and will tackle that task.
```
```suggestion
  in the 30 minutes before smoke testing. Inform that ticking on a testing issue means that someone has assigned themselves and will tackle that task.
```

DanielMurphy22 committed:
Improve clarity

Co-authored-by: Sarah Foxley <55837273+sf1919@users.noreply.github.com>

sf1919 reviewed (approved):
The revisions to the ReleaseChecklist make it so much easier to read. I'm so glad that the days of the week have been removed. I understand whats going on far better!
